### PR TITLE
Incorporate scripts in Jenkins job

### PR DESCRIPTION
**Pull request by KapilPowar**
/runtimes/backlog/issues/1182

- Incorporate scripts into jenkins job.

Signed-off-by: Kapil Powar [kapil.powar@in.ibm.com](mailto:kapil.powar@in.ibm.com)

CC: @llxia

### CODE REVIEW

**Comment by KapilPowar** [comment]
Below are the test job links
[Grinder Test ](https://hyc-runtimes-jenkins.swg-devops.com/job/JCK_Sync/722/) -- PR [#61](https://github.ibm.com/runtimes/JCK8-unzipped/pull/61) created 
[Grinder Test ](https://hyc-runtimes-jenkins.swg-devops.com/job/JCK_Sync/713) -- No updates.

File: buildenv/jenkins/jck_sync [hunk 1]
```diff
@@ -121,6 +125,114 @@ def getJavaSDK(){
...
+                def mkFileContents = readFile(file: mkFileName).trim()
+                if (lastLine && mkFileContents.contains(lastLine)) {
+                    echo "No need to create PR for Tests Group $key"
+                } else {
+                    echo "We need to update $mkFileName and create PR for Tests Group $key"
+                    testClasstList.add(key)
+                    PR_NEEDED=true
+                }
+            }
+        }
```
Comment (llxia) [inline buildenv/jenkins/jck_sync:224]: I believe the logic above should be in a shell script. It calls splitter.sh. Grovvy is for the pipeline.

**Review by llxia — CHANGES_REQUESTED** [review]

**Comment by llxia** [comment]
The general rule for variable names:
- if it is the passed-in job parameter, when reassigning, we keep the same name. For example:
   - `def JCK_GIT_BRANCH = params.JCK_GIT_BRANCH ?: "autoBranch"`, not `GIT_BRANCH`
   -  `def JCK_GIT_BRANCH = params.JCK_GIT_BRANCH ? "-gb ${params.JCK_GIT_BRANCH}": ""` is not recommended as it changes JCK_GIT_BRANCH value with an extra `-gb` and will cause confusion in the code.
- if it is the variables defined in Groovy, we usually use the camel case.
- In general, it is not a good practice to give different variables the same or very similar names.

File: buildenv/jenkins/jck_sync [hunk 2]
```diff
@@ -77,6 +77,10 @@ def syncJckMaterial(){
             }
         }
     }
+    stage('Scripts'){
+        targetFolderCheck(PR_NUMBER)
+        getSplitter(PR_NUMBER)
+    }
```
[the remark this answers is not in the document]
Comment (KapilPowar) [reply buildenv/jenkins/jck_sync:113]: Moved scripts to separate job.

File: buildenv/jenkins/jck_sync (hunk 1 above)
Comment (KapilPowar) [reply buildenv/jenkins/jck_sync]: Yes, logic is moved to shell script.

**Comment by llxia** [comment]
Please separate this PR into several of PRs:
- split subfolders - create `subdirs` dir and move all subdir related *.mk into it. We only need JDK8, 11, 17 and 21.
-  create a standalone job for splitter and folder check
   - there is no need to add the comment about the folder/splitter output in the issue. A PR link is sufficient
   - to avoid re-create the environment and other unnecessary setup, we should use the test target if possible https://github.com/adoptium/aqa-tests/blob/acb19ca9292cf882599f6894b1d5dda22047ae1f/jck/runtime.api/playlist.xml#L32 
   - create test target for splitter in playlist
   - in order to create PR at the aqa-tests, we will need a functional ID. @smlambert do we have a functional ID that we can use? Thanks
- incorporate job for splitter and folder check in sync job

**Comment by KapilPowar** [comment]
Thanks @llxia 
I have created a PR  #4820 for `split subfolders`.
Will update the current PR to create a standalone job for splitter and folder check

**Comment by llxia** [comment]
@KapilPowar Please let me know when this PR is ready for review. Thanks.

**Comment by KapilPowar** [comment]
Moved to internal repo.
Closing it.

**Pull request closed without merging**